[Bug] Simulation with wind speed 0, st dev unchanged crashes with NaN

### Describe the bug

As in the title (originally reported by a user on openrocket-bugs mailing list)

### To Reproduce

1. Open "simple model rocket" example
2. Select a simulation
3. Click "Edit simulation"
4. Set "Average windspeed" to 0
5. Do **not** change standard deviation
6. Close with "OK"
7. Run simulation

Simulation gets an internal exception.

---------- Exception stack trace ----------
net.sf.openrocket.gui.main.SwingExceptionHandler$InternalException: An exception occurred during the simulation
at net.sf.openrocket.gui.main.SwingExceptionHandler.handleErrorCondition(SwingExceptionHandler.java:120)
at net.sf.openrocket.gui.simulation.SimulationRunDialog$InteractiveSimulationWorker.simulationInterrupted(SimulationRunDialog.java:454)
at net.sf.openrocket.gui.simulation.SimulationWorker.done(SimulationWorker.java:109)
at java.desktop/javax.swing.SwingWorker$5.run(SwingWorker.java:750)
at java.desktop/javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.run(SwingWorker.java:848)
at java.desktop/sun.swing.AccumulativeRunnable.run(AccumulativeRunnable.java:112)
at java.desktop/javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.actionPerformed(SwingWorker.java:858)
at java.desktop/javax.swing.Timer.fireActionPerformed(Timer.java:311)
at java.desktop/javax.swing.Timer$DoPostEvent.run(Timer.java:243)
at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:771)
at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:722)
at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:716)
at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:741)
at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:117)
at java.desktop/java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:191)
at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:236)
at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:234)
at java.base/java.security.AccessController.doPrivileged(AccessController.java:318)
at java.desktop/java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:234)
at java.desktop/java.awt.Dialog.show(Dialog.java:1080)
at java.desktop/java.awt.Component.show(Component.java:1728)
at java.desktop/java.awt.Component.setVisible(Component.java:1675)
at java.desktop/java.awt.Window.setVisible(Window.java:1036)
at java.desktop/java.awt.Dialog.setVisible(Dialog.java:1016)
at net.sf.openrocket.gui.main.SimulationPanel.runSimulation(SimulationPanel.java:459)
at net.sf.openrocket.gui.main.SimulationPanel$RunSimulationAction.actionPerformed(SimulationPanel.java:992)
at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1972)
at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2313)
at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405)
at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262)
at java.desktop/javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:279)
at com.github.weisj.darklaf.ui.button.DarkButtonListener.mouseReleased(DarkButtonListener.java:73)
at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:297)
at java.desktop/java.awt.Component.processMouseEvent(Component.java:6626)
at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3389)
at java.desktop/java.awt.Component.processEvent(Component.java:6391)
at java.desktop/java.awt.Container.processEvent(Container.java:2266)
at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5001)
at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324)
at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833)
at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948)
at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575)
at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516)
at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310)
at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2780)
at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833)
at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:773)
at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:722)
at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:716)
at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97)
at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:746)
at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:744)
at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:743)
at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
Caused by: net.sf.openrocket.util.BugException: BUG: Simulation resulted in not-a-number (NaN) value, please report a bug, c=(NaN,NaN,NaN,w=NaN)
at net.sf.openrocket.simulation.AbstractSimulationStepper.checkNaN(AbstractSimulationStepper.java:221)
at net.sf.openrocket.simulation.RK4SimulationStepper.computeParameters(RK4SimulationStepper.java:296)
at net.sf.openrocket.simulation.RK4SimulationStepper.step(RK4SimulationStepper.java:121)
at net.sf.openrocket.simulation.BasicEventSimulationEngine.simulateLoop(BasicEventSimulationEngine.java:171)
at net.sf.openrocket.simulation.BasicEventSimulationEngine.simulate(BasicEventSimulationEngine.java:110)
at net.sf.openrocket.document.Simulation.simulate(Simulation.java:408)
at net.sf.openrocket.gui.simulation.SimulationWorker.doInBackground(SimulationWorker.java:55)
at net.sf.openrocket.gui.simulation.SimulationWorker.doInBackground(SimulationWorker.java:24)
at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)

---------- Thread information ----------
Thread[AWT-EventQueue-0,6,main]

---------- System information ----------
OpenRocket version: 23.09.beta.01
OpenRocket source: default
OpenRocket location: /home/joseph/OpenRocket/OpenRocket.jar
User-defined thrust curves location: /home/joseph/.openrocket/ThrustCurves
LAF: com.github.weisj.darklaf.DarkLaf
JOGL version: 2.5.0-rc-20230523
Current default locale: en_US
System properties:
apple.laf.useScreenMenuBar=true
exe4j.moduleName=/home/joseph/OpenRocket/OpenRocket
file.encoding=UTF-8
file.separator=/
install4j.appDir=/home/joseph/OpenRocket/
install4j.exeDir=/home/joseph/OpenRocket/
install4j.jvmDir=/home/joseph/OpenRocket/jre
install4j.launcherId=59
install4j.swt=false
java.class.path=/home/joseph/OpenRocket/.install4j/i4jruntime.jar:/home/joseph/OpenRocket/.install4j/launcher47cd7162.jar:/home/joseph/OpenRocket/OpenRocket.jar
java.class.version=61.0
java.home=/home/joseph/OpenRocket/jre
java.io.tmpdir=/tmp
java.library.path=/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
java.net.useSystemProxies=true
java.runtime.name=OpenJDK Runtime Environment
java.runtime.version=17.0.7+7-LTS
java.specification.name=Java Platform API Specification
java.specification.vendor=Oracle Corporation
java.specification.version=17
java.util.Arrays.useLegacyMergeSort=true
java.vendor=BellSoft
java.vendor.url=https://bell-sw.com/
java.vendor.url.bug=https://bell-sw.com/support
java.version=17.0.7
java.version.date=2023-04-18
java.vm.compressedOopsMode=Zero based
java.vm.info=mixed mode
java.vm.name=OpenJDK 64-Bit Server VM
java.vm.specification.name=Java Virtual Machine Specification
java.vm.specification.vendor=Oracle Corporation
java.vm.specification.version=17
java.vm.vendor=BellSoft
java.vm.version=17.0.7+7-LTS
jdk.debug=release
jnlp.jogamp.tmp.cache.root=jln10432324184490840038
line.separator=\u000a
native.encoding=UTF-8
os.arch=amd64
os.name=Linux
os.version=6.5.0-1-amd64
path.separator=:
sun.arch.data.model=64
sun.awt.enableExtraMouseButtons=true
sun.awt.exception.handler=net.sf.openrocket.gui.main.SwingExceptionHandler$AwtHandler
sun.boot.library.path=/home/joseph/OpenRocket/jre/lib
sun.cpu.endian=little
sun.font.fontmanager=sun.awt.X11FontManager
sun.io.unicode.encoding=UnicodeLittle
sun.java.command=install4j.net.sf.openrocket.startup.OpenRocket
sun.java.launcher=SUN_STANDARD
sun.jnu.encoding=UTF-8
sun.management.compiler=HotSpot 64-Bit Tiered Compilers
swing.aatext=true
user.country=US
user.dir=/home/joseph/OpenRocket
user.home=/home/joseph
user.language=en
user.name=joseph
user.timezone=America/Denver
---------- Error log ----------
19 0.000 INFO (-) ===== 13 INFO lines removed =====
19 2.372 INFO (MotorDatabaseLoader.java:71) Starting reading user-defined motors
20 2.380 INFO (MotorDatabaseLoader.java:81) Ending reading user-defined motors, motorCount=1900
24 2.874 INFO (SwingStartup.java:226) Opening main application window
28 2.889 INFO (BasicFrame.java:1901) New action initiated
119 3.772 INFO (UpdateInfoRetriever.java:148) Current build is newer
193 8.768 INFO (SwingStartup.java:234) Checking update status
194 8.769 INFO (SwingStartup.java:96) Startup complete
197 13.180 INFO (BasicFrame.java:1273) Opening file from url=jar:file:/home/joseph/OpenRocket/OpenRocket.jar!/datafiles/examples/A simple model rocket.ork filename=A simple model rocket.ork
199 13.185 INFO (BasicFrame.java:1309) Starting OpenFileWorker
200 13.186 INFO (SwingWorkerDialog.java:133) Running SwingWorker net.sf.openrocket.gui.util.OpenFileWorker@ec28c3
201 13.188 INFO (OpenRocketLoader.java:44) Loading .ork file
220 13.472 INFO (OpenRocketLoader.java:96) Loading done
221 13.489 INFO (SwingWorkerDialog.java:151) Worker completed before opening dialog
284 13.845 INFO (BasicFrame.java:1231) Closing window because it is replaceable
320 14.006 INFO (OpenRocketDocument.java:546) Adding undo position 'Modify Rocket' to net.sf.openrocket.document.OpenRocketDocument@185bff27, document was in clean state
402 27.640 USER (DoubleModel.java:111) SpinnerModel setValue called for DoubleModel[SimulationOptions:WindSpeedAverage] newValue=0.0 converted=0.0
415 0.000 WARN (-) ===== 1 WARN lines removed =====
415 27.785 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:WindSpeedDeviation] while still having listeners [javax.swing.JSlider$ModelListener@7dc8d8e0, javax.swing.plaf.basic.BasicSliderUI$Handler@6482fe5b]
422 27.785 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:WindTurbulenceIntensity] while still having listeners [net.sf.openrocket.gui.simulation.SimulationConditionsPanel$1@fd3c195]
426 27.786 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:WindDirection] while still having listeners [javax.swing.JSlider$ModelListener@6b9adb82, javax.swing.plaf.basic.BasicSliderUI$Handler@78c66961]
433 27.786 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchTemperature] while still having listeners [javax.swing.JSlider$ModelListener@630937ae, javax.swing.plaf.basic.BasicSliderUI$Handler@17d748f3]
440 27.786 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchPressure] while still having listeners [javax.swing.JSlider$ModelListener@5fe8449b, javax.swing.plaf.basic.BasicSliderUI$Handler@18bc8e5a]
447 27.786 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchLatitude] while still having listeners [javax.swing.JSlider$ModelListener@3440e3c, javax.swing.plaf.basic.BasicSliderUI$Handler@2f3bb00e]
454 27.787 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchLongitude] while still having listeners [javax.swing.JSlider$ModelListener@b9791c1, javax.swing.plaf.basic.BasicSliderUI$Handler@3255a1ed]
461 27.787 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchAltitude] while still having listeners [javax.swing.JSlider$ModelListener@4b7e893f, javax.swing.plaf.basic.BasicSliderUI$Handler@528bebdf]
468 27.787 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchRodLength] while still having listeners [javax.swing.JSlider$ModelListener@7a4036ef, javax.swing.plaf.basic.BasicSliderUI$Handler@1cdcdb5c]
475 27.788 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchRodAngle] while still having listeners [javax.swing.JSlider$ModelListener@d7c0fc8, javax.swing.plaf.basic.BasicSliderUI$Handler@2380f2cd]
482 27.788 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchRodDirection] while still having listeners [javax.swing.JSlider$ModelListener@1fc05b9, javax.swing.plaf.basic.BasicSliderUI$Handler@2f3bfc93]
489 27.789 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:TimeStep] while still having listeners [javax.swing.JSlider$ModelListener@535d8bea, javax.swing.plaf.basic.BasicSliderUI$Handler@3038e9e1]
497 29.511 INFO (BasicEventSimulationEngine.java:108) >>Starting simulation of branch: Sustainer
502 29.527 INFO (BasicEventSimulationEngine.java:325) Queueing Ignition Event for: Inner Tube/ab4876db / A8 - Armed @: 0.0
508 29.529 INFO (BasicEventSimulationEngine.java:400) Igniting motor: Inner Tube/ab4876db / A8 - Armed @0.0
513 29.624 ERROR (SwingExceptionHandler.java:119) An exception occurred during the simulation
> net.sf.openrocket.util.BugException: BUG: Simulation resulted in not-a-number (NaN) value, please report a bug, c=(NaN,NaN,NaN,w=NaN)
> at net.sf.openrocket.simulation.AbstractSimulationStepper.checkNaN(AbstractSimulationStepper.java:221)
> at net.sf.openrocket.simulation.RK4SimulationStepper.computeParameters(RK4SimulationStepper.java:296)
> at net.sf.openrocket.simulation.RK4SimulationStepper.step(RK4SimulationStepper.java:121)
> at net.sf.openrocket.simulation.BasicEventSimulationEngine.simulateLoop(BasicEventSimulationEngine.java:171)
> at net.sf.openrocket.simulation.BasicEventSimulationEngine.simulate(BasicEventSimulationEngine.java:110)
> at net.sf.openrocket.document.Simulation.simulate(Simulation.java:408)
> at net.sf.openrocket.gui.simulation.SimulationWorker.doInBackground(SimulationWorker.java:55)
> at net.sf.openrocket.gui.simulation.SimulationWorker.doInBackground(SimulationWorker.java:24)
> at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
> at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
> at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
> at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
> at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
> at java.base/java.lang.Thread.run(Thread.java:833)
514 29.624 INFO (SwingExceptionHandler.java:142) Running in EDT, showing dialog
515 29.624 INFO (SwingExceptionHandler.java:201) Showing Exception dialog
516 31.825 USER (SwingExceptionHandler.java:215) User chose not to fill bug report
520 31.843 INFO (SimulationPanel.java:460) Running simulations took 2369 ms
523 33.952 WARN (DoubleModel.java:1006) DoubleModel[constant=1000.0] being garbage-collected while having listeners [net.sf.openrocket.gui.adaptors.DoubleModel$ValueSliderModel@29578487]
524 33.952 WARN (DoubleModel.java:1006) DoubleModel[constant=5.0] being garbage-collected while having listeners [net.sf.openrocket.gui.adaptors.DoubleModel$ValueSliderModel@1524020f]
525 33.953 WARN (DoubleModel.java:1006) DoubleModel[SimulationOptions:WindSpeedAverage] being garbage-collected while having listeners [net.sf.openrocket.gui.adaptors.DoubleModel$ValueSliderModel@4a36e88]
526 33.953 WARN (DoubleModel.java:1006) DoubleModel[constant=0.0] being garbage-collected while having listeners [net.sf.openrocket.gui.adaptors.DoubleModel$ValueSliderModel@4a36e88]
596 39.319 USER (DoubleModel.java:111) SpinnerModel setValue called for DoubleModel[SimulationOptions:WindSpeedAverage] newValue=0.0 converted=0.0
600 42.462 USER (DoubleModel.java:111) SpinnerModel setValue called for DoubleModel[SimulationOptions:WindSpeedDeviation] newValue=0.0 converted=0.0
606 42.611 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:WindSpeedAverage] while still having listeners [javax.swing.JSlider$ModelListener@e99b2e2, javax.swing.plaf.basic.BasicSliderUI$Handler@32bcadb7]
613 42.611 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:WindSpeedDeviation] while still having listeners [javax.swing.JSlider$ModelListener@42a8ea7, javax.swing.plaf.basic.BasicSliderUI$Handler@45dac9db]
620 42.611 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:WindTurbulenceIntensity] while still having listeners [net.sf.openrocket.gui.simulation.SimulationConditionsPanel$1@552edf5b]
624 42.611 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:WindDirection] while still having listeners [javax.swing.JSlider$ModelListener@137e3c04, javax.swing.plaf.basic.BasicSliderUI$Handler@5b6d0311]
631 42.612 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchTemperature] while still having listeners [javax.swing.JSlider$ModelListener@1dd8af4, javax.swing.plaf.basic.BasicSliderUI$Handler@3e28b601]
638 42.612 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchPressure] while still having listeners [javax.swing.JSlider$ModelListener@1a80302, javax.swing.plaf.basic.BasicSliderUI$Handler@1ef52d1d]
645 42.612 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchLatitude] while still having listeners [javax.swing.JSlider$ModelListener@2c224aac, javax.swing.plaf.basic.BasicSliderUI$Handler@399305e7]
652 42.614 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchLongitude] while still having listeners [javax.swing.JSlider$ModelListener@5e7a0c87, javax.swing.plaf.basic.BasicSliderUI$Handler@1c89d9a5]
659 42.615 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchAltitude] while still having listeners [javax.swing.JSlider$ModelListener@1e4e4487, javax.swing.plaf.basic.BasicSliderUI$Handler@78ba4232]
666 42.615 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchRodLength] while still having listeners [javax.swing.JSlider$ModelListener@4cfa2945, javax.swing.plaf.basic.BasicSliderUI$Handler@2f720311]
673 42.615 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchRodAngle] while still having listeners [javax.swing.JSlider$ModelListener@44174791, javax.swing.plaf.basic.BasicSliderUI$Handler@3692fc60]
680 42.616 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchRodDirection] while still having listeners [javax.swing.JSlider$ModelListener@614a3f3a, javax.swing.plaf.basic.BasicSliderUI$Handler@44c23984]
687 42.616 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:TimeStep] while still having listeners [javax.swing.JSlider$ModelListener@6335f3fd, javax.swing.plaf.basic.BasicSliderUI$Handler@2feeac9d]
694 44.967 INFO (BasicEventSimulationEngine.java:108) >>Starting simulation of branch: Sustainer
698 44.968 INFO (BasicEventSimulationEngine.java:325) Queueing Ignition Event for: Inner Tube/ab4876db / B4 - Armed @: 0.0
704 44.968 INFO (BasicEventSimulationEngine.java:400) Igniting motor: Inner Tube/ab4876db / B4 - Armed @0.0
739 45.024 WARN (DoubleModel.java:1006) DoubleModel[constant=0.0] being garbage-collected while having listeners [net.sf.openrocket.gui.adaptors.DoubleModel$ValueSliderModel@458e83d7]
740 45.024 WARN (DoubleModel.java:1006) DoubleModel[SimulationOptions:WindSpeedAverage] being garbage-collected while having listeners [net.sf.openrocket.gui.adaptors.DoubleModel$ValueSliderModel@458e83d7]
741 45.024 WARN (DoubleModel.java:1006) DoubleModel[constant=5.0] being garbage-collected while having listeners [net.sf.openrocket.gui.adaptors.DoubleModel$ValueSliderModel@5fd79b30]
742 45.024 WARN (DoubleModel.java:1006) DoubleModel[constant=1000.0] being garbage-collected while having listeners [net.sf.openrocket.gui.adaptors.DoubleModel$ValueSliderModel@605c8dc8]
1599 45.300 INFO (BasicEventSimulationEngine.java:108) >>Starting simulation of branch: Sustainer
1603 45.303 INFO (BasicEventSimulationEngine.java:325) Queueing Ignition Event for: Inner Tube/ab4876db / B4 - Armed @: 0.0
1609 45.304 INFO (BasicEventSimulationEngine.java:400) Igniting motor: Inner Tube/ab4876db / B4 - Armed @0.0
2599 45.502 INFO (BasicEventSimulationEngine.java:114) <<Finished simulating branch: Sustainer curTime:5.4531445312499995 finTime:5.4531445312499995
3265 45.534 INFO (BasicEventSimulationEngine.java:114) <<Finished simulating branch: Sustainer curTime:38.32998367488419 finTime:38.32998367488419
3277 45.548 INFO (SimulationPanel.java:460) Running simulations took 586 ms
3351 253.148 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:WindSpeedAverage] while still having listeners [javax.swing.JSlider$ModelListener@6be25bcf, javax.swing.plaf.basic.BasicSliderUI$Handler@5c8fd8b]
3358 253.148 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:WindSpeedDeviation] while still having listeners [javax.swing.JSlider$ModelListener@4247c7d, javax.swing.plaf.basic.BasicSliderUI$Handler@51e35197]
3365 253.149 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:WindTurbulenceIntensity] while still having listeners [net.sf.openrocket.gui.simulation.SimulationConditionsPanel$1@292612fc]
3369 253.149 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:WindDirection] while still having listeners [javax.swing.JSlider$ModelListener@14b01957, javax.swing.plaf.basic.BasicSliderUI$Handler@3f360a18]
3376 253.149 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchTemperature] while still having listeners [javax.swing.JSlider$ModelListener@424ac4be, javax.swing.plaf.basic.BasicSliderUI$Handler@212512f5]
3383 253.149 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchPressure] while still having listeners [javax.swing.JSlider$ModelListener@64e69534, javax.swing.plaf.basic.BasicSliderUI$Handler@35eddbae]
3390 253.149 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchLatitude] while still having listeners [javax.swing.JSlider$ModelListener@2fbd0ee0, javax.swing.plaf.basic.BasicSliderUI$Handler@5a58585c]
3397 253.150 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchLongitude] while still having listeners [javax.swing.JSlider$ModelListener@4cf30d8c, javax.swing.plaf.basic.BasicSliderUI$Handler@4592651e]
3404 253.150 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchAltitude] while still having listeners [javax.swing.JSlider$ModelListener@1f906fea, javax.swing.plaf.basic.BasicSliderUI$Handler@65f9630f]
3411 253.150 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchRodLength] while still having listeners [javax.swing.JSlider$ModelListener@3290d215, javax.swing.plaf.basic.BasicSliderUI$Handler@2a74ae59]
3418 253.150 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchRodAngle] while still having listeners [javax.swing.JSlider$ModelListener@45b59500, javax.swing.plaf.basic.BasicSliderUI$Handler@3f032e4e]
3425 253.150 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:LaunchRodDirection] while still having listeners [javax.swing.JSlider$ModelListener@749ecbff, javax.swing.plaf.basic.BasicSliderUI$Handler@6b31b14c]
3432 253.152 WARN (DoubleModel.java:990) Invalidating DoubleModel[SimulationOptions:TimeStep] while still having listeners [javax.swing.JSlider$ModelListener@b0210ee, javax.swing.plaf.basic.BasicSliderUI$Handler@1141bb59]
3439 255.781 INFO (BasicEventSimulationEngine.java:108) >>Starting simulation of branch: Sustainer
3443 255.783 INFO (BasicEventSimulationEngine.java:325) Queueing Ignition Event for: Inner Tube/ab4876db / B4 - Armed @: 0.0
3449 255.784 INFO (BasicEventSimulationEngine.java:400) Igniting motor: Inner Tube/ab4876db / B4 - Armed @0.0
4179 255.920 WARN (DoubleModel.java:1006) DoubleModel[constant=0.0] being garbage-collected while having listeners [net.sf.openrocket.gui.adaptors.DoubleModel$ValueSliderModel@2e45c375]
4180 255.920 WARN (DoubleModel.java:1006) DoubleModel[SimulationOptions:WindSpeedAverage] being garbage-collected while having listeners [net.sf.openrocket.gui.adaptors.DoubleModel$ValueSliderModel@2e45c375]
4181 255.920 WARN (DoubleModel.java:1006) DoubleModel[constant=5.0] being garbage-collected while having listeners [net.sf.openrocket.gui.adaptors.DoubleModel$ValueSliderModel@4b723f1b]
4182 255.920 WARN (DoubleModel.java:1006) DoubleModel[constant=1000.0] being garbage-collected while having listeners [net.sf.openrocket.gui.adaptors.DoubleModel$ValueSliderModel@2a514f70]
4339 255.943 INFO (BasicEventSimulationEngine.java:108) >>Starting simulation of branch: Sustainer
4343 255.943 INFO (BasicEventSimulationEngine.java:325) Queueing Ignition Event for: Inner Tube/ab4876db / B4 - Armed @: 0.0
4349 255.944 INFO (BasicEventSimulationEngine.java:400) Igniting motor: Inner Tube/ab4876db / B4 - Armed @0.0
4549 0.000 DEBUG (-) ===== 300 DEBUG lines removed =====
4549 255.977 DEBUG (SimulationRunDialog.java:242) Progressbar value 0
4652 255.991 DEBUG (BasicEventSimulationEngine.java:449) detected Motor Burnout for motor B4@ 1.03 on stage 0: Sustainer
4897 256.021 DEBUG (SimulationRunDialog.java:354) updating simulationMaxAltitude[0] to 19.375287378450054
4898 256.021 DEBUG (SimulationRunDialog.java:366) simulationStage boost: t=0.7535923813553698 est=0.9963345279170399
4899 256.021 DEBUG (SimulationRunDialog.java:464) Setting progress to 30 (real 30)
4900 256.021 DEBUG (SimulationRunDialog.java:242) Progressbar value 30
5329 256.078 DEBUG (SimulationRunDialog.java:482) APOGEE
5333 256.078 DEBUG (SimulationRunDialog.java:492) END
5338 256.078 DEBUG (FlightData.java:255) Computed flight values: maxAltitude=136.6830225033572 maxVelocity=53.30700602317454 maxAcceleration=180.4963934832766 maxMachNumber=0.15666511229170524 timeToApogee=5.40314453125 flightTime=5.4531445312499995 groundHitVelocity=NaN launchRodVelocity=18.22347194534186
5339 256.078 INFO (BasicEventSimulationEngine.java:114) <<Finished simulating branch: Sustainer curTime:5.4531445312499995 finTime:5.4531445312499995
5340 256.078 DEBUG (BasicEventSimulationEngine.java:577) deployed recovery devices: 1
5420 256.080 DEBUG (SimulationRunDialog.java:482) APOGEE
5969 0.000 VBOSE (-) ===== 5551 VBOSE lines removed =====
5969 256.097 VBOSE (BasicEventSimulationEngine.java:300) Remaining EventQueue = []
5970 256.097 VBOSE (BasicEventSimulationEngine.java:383) Handling event FlightEvent[type=ALTITUDE,time=37.27187790534698,source=A simple model rocket,data=[6.470778362192782;4.394199577417195]]
5971 256.097 VBOSE (BasicEventSimulationEngine.java:170) Taking simulation step at t=37.27187790534698 altitude 4.394199577417195
5972 256.097 VBOSE (AbstractEulerStepper.java:90) timeStep is 0.5
5973 256.097 VBOSE (AbstractEulerStepper.java:185) time 37.77187790534698, altitude 2.3111409421255136, velocity -4.2169438474416125
5974 256.097 VBOSE (BasicEventSimulationEngine.java:297) HandleEvents: current branch = Sustainer
5975 256.097 VBOSE (BasicEventSimulationEngine.java:299) Obtained event from queue: FlightEvent[type=ALTITUDE,time=37.77187790534698,source=A simple model rocket,data=[4.394199577417195;2.3111409421255136]]
5976 256.097 VBOSE (BasicEventSimulationEngine.java:300) Remaining EventQueue = []
5977 256.097 VBOSE (BasicEventSimulationEngine.java:383) Handling event FlightEvent[type=ALTITUDE,time=37.77187790534698,source=A simple model rocket,data=[4.394199577417195;2.3111409421255136]]
5978 256.097 VBOSE (BasicEventSimulationEngine.java:170) Taking simulation step at t=37.77187790534698 altitude 2.3111409421255136
5979 256.097 VBOSE (AbstractEulerStepper.java:90) timeStep is 0.5
5980 256.098 VBOSE (AbstractEulerStepper.java:185) time 38.27187790534698, altitude 0.2375331180840452, velocity -4.077487448724261
5981 256.098 VBOSE (BasicEventSimulationEngine.java:297) HandleEvents: current branch = Sustainer
5982 256.098 VBOSE (BasicEventSimulationEngine.java:299) Obtained event from queue: FlightEvent[type=ALTITUDE,time=38.27187790534698,source=A simple model rocket,data=[2.3111409421255136;0.2375331180840452]]
5983 256.098 VBOSE (BasicEventSimulationEngine.java:300) Remaining EventQueue = []
5984 256.098 VBOSE (BasicEventSimulationEngine.java:383) Handling event FlightEvent[type=ALTITUDE,time=38.27187790534698,source=A simple model rocket,data=[2.3111409421255136;0.2375331180840452]]
5985 256.098 VBOSE (BasicEventSimulationEngine.java:170) Taking simulation step at t=38.27187790534698 altitude 0.2375331180840452
5986 256.098 VBOSE (AbstractEulerStepper.java:90) timeStep is 0.5
5987 256.098 VBOSE (AbstractEulerStepper.java:106) ground hit changes timeStep to 0.058099438384127164
5988 256.098 VBOSE (AbstractEulerStepper.java:185) time 38.32997734373111, altitude 0.0, velocity -4.099291008705404
5989 256.098 VBOSE (BasicEventSimulationEngine.java:297) HandleEvents: current branch = Sustainer
5990 256.098 VBOSE (BasicEventSimulationEngine.java:299) Obtained event from queue: FlightEvent[type=GROUND_HIT,time=38.32997734373111,source=null,data=null]
5991 256.098 VBOSE (BasicEventSimulationEngine.java:300) Remaining EventQueue = [FlightEvent[type=ALTITUDE,time=38.32997734373111,source=A simple model rocket,data=[0.2375331180840452;0.0]]]
5992 256.098 VBOSE (BasicEventSimulationEngine.java:383) Handling event FlightEvent[type=GROUND_HIT,time=38.32997734373111,source=null,data=null]
5993 256.098 VBOSE (GroundStepper.java:16) initializing GroundStepper
5994 256.098 VBOSE (BasicEventSimulationEngine.java:299) Obtained event from queue: FlightEvent[type=ALTITUDE,time=38.32997734373111,source=A simple model rocket,data=[0.2375331180840452;0.0]]
5995 256.098 VBOSE (BasicEventSimulationEngine.java:300) Remaining EventQueue = []
5996 256.098 VBOSE (BasicEventSimulationEngine.java:383) Handling event FlightEvent[type=ALTITUDE,time=38.32997734373111,source=A simple model rocket,data=[0.2375331180840452;0.0]]
5997 256.098 VBOSE (BasicEventSimulationEngine.java:170) Taking simulation step at t=38.32997734373111 altitude 0.0
5998 256.098 VBOSE (GroundStepper.java:22) step: position=(-2.56348,-0.00078,0.00000), velocity=(-0.10000,-0.00000,-4.09929)
5999 256.098 VBOSE (BasicEventSimulationEngine.java:297) HandleEvents: current branch = Sustainer
6000 256.098 VBOSE (BasicEventSimulationEngine.java:299) Obtained event from queue: FlightEvent[type=SIMULATION_END,time=38.32997734373111,source=null,data=null]
6001 256.098 VBOSE (BasicEventSimulationEngine.java:300) Remaining EventQueue = []
6002 256.098 DEBUG (SimulationRunDialog.java:492) END
6003 256.098 VBOSE (BasicEventSimulationEngine.java:383) Handling event FlightEvent[type=SIMULATION_END,time=38.32997734373111,source=null,data=null]
6004 256.098 DEBUG (FlightData.java:255) Computed flight values: maxAltitude=136.51266021455288 maxVelocity=53.30700602317454 maxAcceleration=180.4963934832766 maxMachNumber=0.15666511229170524 timeToApogee=5.315189732379825 flightTime=38.32997734373111 groundHitVelocity=4.099291008705404 launchRodVelocity=18.22347194534186
6005 256.098 INFO (BasicEventSimulationEngine.java:114) <<Finished simulating branch: Sustainer curTime:38.32997734373111 finTime:38.32997734373111
6006 256.098 DEBUG (Simulation.java:410) Simulation: returning from simulator, simulation took 318ms
6007 256.111 DEBUG (SimulationRunDialog.java:354) updating simulationMaxAltitude[0] to 136.66591720759797
6008 256.111 DEBUG (SimulationRunDialog.java:354) updating simulationMaxAltitude[0] to 136.04836718425452
6009 256.111 DEBUG (SimulationRunDialog.java:354) updating simulationMaxAltitude[0] to 136.35652594409893
6010 256.111 DEBUG (SimulationRunDialog.java:377) CHANGING to simulationStage -1, vel=0.1
6011 256.111 DEBUG (SimulationRunDialog.java:393) CHANGING to simulationStage 0, apogee=136.35652594409893
6012 256.111 DEBUG (SimulationRunDialog.java:398) simulationStage landing (0): alt=0.0 apogee=136.35652594409893
6013 256.111 DEBUG (SimulationRunDialog.java:464) Setting progress to 100 (real 100)
6014 256.111 DEBUG (SimulationRunDialog.java:242) Progressbar value 100
6015 256.112 DEBUG (SimulationRunDialog.java:409) Simulation done
6016 256.112 DEBUG (SimulationRunDialog.java:464) Setting progress to 100 (real 100)
6017 256.112 DEBUG (SimulationRunDialog.java:230) Everything done.
6018 256.118 INFO (SimulationPanel.java:460) Running simulations took 339 ms
6019 256.131 DEBUG (GUIUtil.java:272) Clearing all models of window net.sf.openrocket.gui.simulation.SimulationRunDialog[dialog7,623,611,341x214,invalid,hidden,layout=java.awt.BorderLayout,APPLICATION_MODAL,title=Running simulations…,defaultCloseOperation=DISPOSE_ON_CLOSE,rootPane=javax.swing.JRootPane[,6,29,329x179,invalid,layout=javax.swing.JRootPane$RootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=16777673,maximumSize=,minimumSize=,preferredSize=],rootPaneCheckingEnabled=true]
6020 256.131 DEBUG (MemoryManagement.java:56) Adding object into collectable list: net.sf.openrocket.gui.simulation.SimulationRunDialog[dialog7,623,611,341x214,invalid,hidden,layout=java.awt.BorderLayout,APPLICATION_MODAL,title=Running simulations…,defaultCloseOperation=DISPOSE_ON_CLOSE,rootPane=javax.swing.JRootPane[,6,29,329x179,invalid,layout=javax.swing.JRootPane$RootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=16777673,maximumSize=,minimumSize=,preferredSize=],rootPaneCheckingEnabled=true]
6021 260.531 DEBUG (Simulation.java:406) Simulation: calling simulator
6022 260.532 INFO (BasicEventSimulationEngine.java:108) >>Starting simulation of branch: Sustainer
6023 260.533 VBOSE (BasicEventSimulationEngine.java:297) HandleEvents: current branch = Sustainer
6024 260.533 VBOSE (BasicEventSimulationEngine.java:299) Obtained event from queue: FlightEvent[type=LAUNCH,time=0.0,source=A simple model rocket,data=null]
6025 260.533 VBOSE (BasicEventSimulationEngine.java:300) Remaining EventQueue = []
6026 260.533 INFO (BasicEventSimulationEngine.java:325) Queueing Ignition Event for: Inner Tube/ab4876db / A8 - Armed @: 0.0
6027 260.533 DEBUG (SimulationRunDialog.java:487) LAUNCH
6028 260.533 VBOSE (BasicEventSimulationEngine.java:383) Handling event FlightEvent[type=LAUNCH,time=0.0,source=A simple model rocket,data=null]
6029 260.533 VBOSE (BasicEventSimulationEngine.java:299) Obtained event from queue: FlightEvent[type=IGNITION,time=0.0,source=Inner Tube,data=A8]
6030 260.533 VBOSE (BasicEventSimulationEngine.java:300) Remaining EventQueue = []
6031 260.533 VBOSE (BasicEventSimulationEngine.java:383) Handling event FlightEvent[type=IGNITION,time=0.0,source=Inner Tube,data=A8]
6032 260.533 INFO (BasicEventSimulationEngine.java:400) Igniting motor: Inner Tube/ab4876db / A8 - Armed @0.0
6033 260.533 VBOSE (BasicEventSimulationEngine.java:170) Taking simulation step at t=0.0 altitude 0.0
6034 260.535 DEBUG (SimulationRunDialog.java:242) Progressbar value 0
6035 260.535 DEBUG (SimulationRunDialog.java:247) No simulation status data available, setting empty labels
6036 260.622 DEBUG (SimulationRunDialog.java:366) simulationStage boost: t=0.0 est=0.6630044035350177
6037 260.622 DEBUG (SimulationRunDialog.java:464) Setting progress to 0 (real 0)
6038 260.622 DEBUG (SimulationRunDialog.java:242) Progressbar value 0
6039 260.622 ERROR (SwingExceptionHandler.java:119) An exception occurred during the simulation
> net.sf.openrocket.util.BugException: BUG: Simulation resulted in not-a-number (NaN) value, please report a bug, c=(NaN,NaN,NaN,w=NaN)
> at net.sf.openrocket.simulation.AbstractSimulationStepper.checkNaN(AbstractSimulationStepper.java:221)
> at net.sf.openrocket.simulation.RK4SimulationStepper.computeParameters(RK4SimulationStepper.java:296)
> at net.sf.openrocket.simulation.RK4SimulationStepper.step(RK4SimulationStepper.java:121)
> at net.sf.openrocket.simulation.BasicEventSimulationEngine.simulateLoop(BasicEventSimulationEngine.java:171)
> at net.sf.openrocket.simulation.BasicEventSimulationEngine.simulate(BasicEventSimulationEngine.java:110)
> at net.sf.openrocket.document.Simulation.simulate(Simulation.java:408)
> at net.sf.openrocket.gui.simulation.SimulationWorker.doInBackground(SimulationWorker.java:55)
> at net.sf.openrocket.gui.simulation.SimulationWorker.doInBackground(SimulationWorker.java:24)
> at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
> at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
> at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
> at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
> at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
> at java.base/java.lang.Thread.run(Thread.java:833)
6040 260.622 INFO (SwingExceptionHandler.java:142) Running in EDT, showing dialog
6041 260.622 INFO (SwingExceptionHandler.java:201) Showing Exception dialog
6042 262.138 USER (SwingExceptionHandler.java:220) User requested sending bug report
6043 262.142 DEBUG (JarUtil.java:40) Found jar file using codeSource
6044 303.972 DEBUG (SimulationRunDialog.java:409) Simulation done
6045 303.972 DEBUG (SimulationRunDialog.java:464) Setting progress to 100 (real 100)
6046 303.972 DEBUG (SimulationRunDialog.java:230) Everything done.
6047 303.983 INFO (SimulationPanel.java:460) Running simulations took 43454 ms
6048 303.985 DEBUG (GUIUtil.java:272) Clearing all models of window net.sf.openrocket.gui.dialogs.BugReportDialog[dialog8,252,0,838x1049,invalid,hidden,layout=java.awt.BorderLayout,APPLICATION_MODAL,title=Bug report,defaultCloseOperation=DISPOSE_ON_CLOSE,rootPane=javax.swing.JRootPane[,6,29,826x1014,invalid,layout=javax.swing.JRootPane$RootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=16777673,maximumSize=,minimumSize=,preferredSize=],rootPaneCheckingEnabled=true]
6049 303.985 DEBUG (MemoryManagement.java:56) Adding object into collectable list: net.sf.openrocket.gui.dialogs.BugReportDialog[dialog8,252,0,838x1049,invalid,hidden,layout=java.awt.BorderLayout,APPLICATION_MODAL,title=Bug report,defaultCloseOperation=DISPOSE_ON_CLOSE,rootPane=javax.swing.JRootPane[,6,29,826x1014,invalid,layout=javax.swing.JRootPane$RootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=16777673,maximumSize=,minimumSize=,preferredSize=],rootPaneCheckingEnabled=true]
6050 303.986 DEBUG (GUIUtil.java:272) Clearing all models of window net.sf.openrocket.gui.simulation.SimulationRunDialog[dialog9,623,611,341x214,invalid,hidden,layout=java.awt.BorderLayout,APPLICATION_MODAL,title=Running simulations…,defaultCloseOperation=DISPOSE_ON_CLOSE,rootPane=javax.swing.JRootPane[,6,29,329x179,invalid,layout=javax.swing.JRootPane$RootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=16777673,maximumSize=,minimumSize=,preferredSize=],rootPaneCheckingEnabled=true]
6051 303.986 DEBUG (MemoryManagement.java:56) Adding object into collectable list: net.sf.openrocket.gui.simulation.SimulationRunDialog[dialog9,623,611,341x214,invalid,hidden,layout=java.awt.BorderLayout,APPLICATION_MODAL,title=Running simulations…,defaultCloseOperation=DISPOSE_ON_CLOSE,rootPane=javax.swing.JRootPane[,6,29,329x179,invalid,layout=javax.swing.JRootPane$RootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=16777673,maximumSize=,minimumSize=,preferredSize=],rootPaneCheckingEnabled=true]
6052 307.166 DEBUG (Simulation.java:406) Simulation: calling simulator
6053 307.167 INFO (BasicEventSimulationEngine.java:108) >>Starting simulation of branch: Sustainer
6054 307.168 VBOSE (BasicEventSimulationEngine.java:297) HandleEvents: current branch = Sustainer
6055 307.168 VBOSE (BasicEventSimulationEngine.java:299) Obtained event from queue: FlightEvent[type=LAUNCH,time=0.0,source=A simple model rocket,data=null]
6056 307.168 VBOSE (BasicEventSimulationEngine.java:300) Remaining EventQueue = []
6057 307.168 INFO (BasicEventSimulationEngine.java:325) Queueing Ignition Event for: Inner Tube/ab4876db / A8 - Armed @: 0.0
6058 307.168 DEBUG (SimulationRunDialog.java:487) LAUNCH
6059 307.168 VBOSE (BasicEventSimulationEngine.java:383) Handling event FlightEvent[type=LAUNCH,time=0.0,source=A simple model rocket,data=null]
6060 307.168 VBOSE (BasicEventSimulationEngine.java:299) Obtained event from queue: FlightEvent[type=IGNITION,time=0.0,source=Inner Tube,data=A8]
6061 307.168 VBOSE (BasicEventSimulationEngine.java:300) Remaining EventQueue = []
6062 307.168 VBOSE (BasicEventSimulationEngine.java:383) Handling event FlightEvent[type=IGNITION,time=0.0,source=Inner Tube,data=A8]
6063 307.168 INFO (BasicEventSimulationEngine.java:400) Igniting motor: Inner Tube/ab4876db / A8 - Armed @0.0
6064 307.168 VBOSE (BasicEventSimulationEngine.java:170) Taking simulation step at t=0.0 altitude 0.0
6065 307.173 DEBUG (SimulationRunDialog.java:242) Progressbar value 0
6066 307.173 DEBUG (SimulationRunDialog.java:247) No simulation status data available, setting empty labels
6067 307.260 DEBUG (SimulationRunDialog.java:366) simulationStage boost: t=0.0 est=0.6630044035350177
6068 307.260 DEBUG (SimulationRunDialog.java:464) Setting progress to 0 (real 0)
6069 307.260 DEBUG (SimulationRunDialog.java:242) Progressbar value 0
6070 307.260 ERROR (SwingExceptionHandler.java:119) An exception occurred during the simulation
> net.sf.openrocket.util.BugException: BUG: Simulation resulted in not-a-number (NaN) value, please report a bug, c=(NaN,NaN,NaN,w=NaN)
> at net.sf.openrocket.simulation.AbstractSimulationStepper.checkNaN(AbstractSimulationStepper.java:221)
> at net.sf.openrocket.simulation.RK4SimulationStepper.computeParameters(RK4SimulationStepper.java:296)
> at net.sf.openrocket.simulation.RK4SimulationStepper.step(RK4SimulationStepper.java:121)
> at net.sf.openrocket.simulation.BasicEventSimulationEngine.simulateLoop(BasicEventSimulationEngine.java:171)
> at net.sf.openrocket.simulation.BasicEventSimulationEngine.simulate(BasicEventSimulationEngine.java:110)
> at net.sf.openrocket.document.Simulation.simulate(Simulation.java:408)
> at net.sf.openrocket.gui.simulation.SimulationWorker.doInBackground(SimulationWorker.java:55)
> at net.sf.openrocket.gui.simulation.SimulationWorker.doInBackground(SimulationWorker.java:24)
> at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
> at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
> at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
> at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
> at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
> at java.base/java.lang.Thread.run(Thread.java:833)
6071 307.261 INFO (SwingExceptionHandler.java:142) Running in EDT, showing dialog
6072 307.261 INFO (SwingExceptionHandler.java:201) Showing Exception dialog
6073 336.539 USER (SwingExceptionHandler.java:220) User requested sending bug report
6074 336.541 DEBUG (JarUtil.java:40) Found jar file using codeSource
---------- End of bug report ----------

### Screenshots / .ork file

#### Screenshot(s):
*(drag-and-drop the screenshot(s) here)*


#### .ork file:
*(drag-and-drop the file here as a .zip file)*


### OpenRocket version

22.02, 23.09

### What platform are you running on?

Linux

### Additional context

The issue was masked if the st dev was changed, as setWindSpeedDeviation() in SimulationOptions checks the wind speed and ensures it is at least 0.1. The UI is not updated to reflect this(!). You can manually set the wind speed to 0, and as long as you don't update the standard deviation it stays as set.

1. Note that a wind speed of 0 can't have a st dev other than 0
2. We appear to be using a pink noise generator for our wind model. Looking briefly at wind speed distributions, we should probably be using something else (and I'd never heard of the ones I'm seeing -- Weibull seems to be popular?)

--- a/core/src/net/sf/openrocket/aerodynamics/barrowman/TubeCalc.java
+++ b/core/src/net/sf/openrocket/aerodynamics/barrowman/TubeCalc.java
@@ -40,6 +40,11 @@ public abstract class TubeCalc extends RocketComponentCalc {
 	public double calculatePressureCD(FlightConditions conditions,
 			double stagnationCD, double baseCD, WarningSet warnings) {
 
+		// If we aren't moving, treat CD as 0
+		final double v = conditions.getVelocity();
+		if (v < MathUtil.EPSILON)
+			return 0;
+		
 		// Need to check for tube inner area 0 in case of rockets using launch lugs with
 		// an inner radius of 0 to emulate rail guides (or just weird rockets, of course)
 		double tubeCD = 0.0;
@@ -49,7 +54,6 @@ public abstract class TubeCalc extends RocketComponentCalc {
 			final double p = conditions.getAtmosphericConditions().getPressure();
 			final double t = conditions.getAtmosphericConditions().getTemperature();
 			final double rho = conditions.getAtmosphericConditions().getDensity();
-			final double v = conditions.getVelocity();
 			
 			// Reynolds number (note Reynolds number for the interior of a pipe is based on diameter,
 			// not length (t))

